tests/Unit/CrudPanel/CrudPanelTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Backpack\CRUD\Tests\Unit\CrudPanel;
+
+use Illuminate\Database\Eloquent\Builder;
+
+class CrudPanelTest extends BaseCrudPanelTest
+{
+    public function testSetModelFromModelClass()
+    {
+        $this->crudPanel->setModel(TestModel::class);
+
+        $this->assertEquals($this->model, $this->crudPanel->model);
+        $this->assertInstanceOf(TestModel::class, $this->crudPanel->model);
+        $this->assertInstanceOf(Builder::class, $this->crudPanel->query);
+    }
+
+    public function testSetModelFromModelClassName()
+    {
+        $this->crudPanel->setModel('\Backpack\CRUD\Tests\Unit\CrudPanel\TestModel');
+
+        $this->assertEquals($this->model, $this->crudPanel->model);
+        $this->assertInstanceOf('\Backpack\CRUD\Tests\Unit\CrudPanel\TestModel', $this->crudPanel->model);
+        $this->assertInstanceOf(Builder::class, $this->crudPanel->query);
+    }
+
+    public function testSetUnknownModel()
+    {
+        $this->setExpectedException(\Exception::class);
+
+        $this->crudPanel->setModel('\Foo\Bar');
+    }
+
+    public function testSetRouteName()
+    {
+        // TODO: check if we need an app instance to test this out
+        $this->markTestIncomplete();
+    }
+
+    public function testSetUnknownRouteName()
+    {
+        $this->setExpectedException(\Exception::class);
+
+        $this->crudPanel->setRouteName('unknown.route.name');
+    }
+
+    public function testSync()
+    {
+        $this->markTestIncomplete();
+
+        // TODO: the sync method should not be in the CrudPanel class and should not be exposed in the public API.
+        //       it is an utility method and should be refactored.
```

Comment:
`a utility` is more idiomatic English :imp: and the `i` should be capitalised.

Reply:
Ok, will update.